virtual_rainforest/core/config.py

```diff
@@ -500,31 +521,24 @@ def load_config_toml_string(self) -> None:
             ConfigurationError: Invalid TOML string.
         """
 
-        # Check for a string to parse
-        if self.cfg_string is None:
-            to_raise: Exception = RuntimeError(
-                "Cannot run load_config_toml_string: no cfg_string loaded."
-            )
-            LOGGER.critical(to_raise)
-            raise to_raise
+        for index, cfg_string in enumerate(self.cfg_strings):
+            # Load the contents into the instance
+            try:
+                self.toml_contents[f"cfg_string_{index}"] = tomllib.loads(cfg_string)
+            except tomllib.TOMLDecodeError as err:
+                to_raise = ConfigurationError(
+                    f"TOML parsing error in cfg_strings: {str(err)}"
```

Comment:
I just learned you can do this with f-strings:

```suggestion
                    f"TOML parsing error in cfg_strings: {err!s}"
```

Reply:
I guess there's always one more formatting syntax! I think the existing code is clearer, but that might just be ignorance of the format tricks.

Reply:
Yeah I don't think it's much clearer. But I thought I'd show you the trick anyway :stuck_out_tongue:. You can also write `{thing!r}` instead of `{repr(thing)}`.